### DISCUSSION TITLE
Lesson 40. Raffle deployment into Sepolia

### DISCUSSION THREAD

**Discussion by szopson** [Cyfrin/foundry-full-course-cu#2126] [Q&A] [answered]
I cant  deploy the contract into sepolia network where I am getting AddConsumer error. The one below:
`[⠒] Compiling...
No files changed, compilation skipped
Traces:
  [21131893] → new DeployRaffle@0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519
    └─ ← [Return] 105344 bytes of code

  [11127540] DeployRaffle::run()
    ├─ [4537148] → new HelperConfig@0xC7f2Cf4845C6db0e1a1e91ED41Bcd0FcC1b0E141
    │   └─ ← [Return] 21442 bytes of code
    ├─ [2312] HelperConfig::getConfig()
    │   └─ ← [Return] NetworkConfig({ entranceFee: 10000000000000000 [1e16], interval: 30, vrfCoordinator: 0x8103B0A8A00be2DDC778e6e7eaa21791Cd364625, gasLane: 0x8077df514608a09f83e4e8d300645594e5d7234665448ba83f51a50f842bd3d9, callbackGasLimit: 500000 [5e5], subscriptionId: 73878881164087465189628802004031071946213112452267996662201529735988893001902 [7.387e76], link: 0x779877A7B0D9E8603169DdbD7836e478b4624789, account: 0x66C6e902bA95dAD75e84152361A210b3c7Dc1e9f })
    ├─ [0] VM::startBroadcast(0x66C6e902bA95dAD75e84152361A210b3c7Dc1e9f)
    │   └─ ← [Return] 
    ├─ [826464] → new Raffle@0xe9D52cB94987eB686351B642479C688797A54cb9
    │   └─ ← [Return] 3782 bytes of code
    ├─ [0] VM::stopBroadcast()
    │   └─ ← [Return] 
    ├─ [5642148] → new AddConsumer@0xdaE97900D4B184c5D2012dcdB658c008966466DD
    │   └─ ← [Return] 28066 bytes of code
    ├─ [8753] AddConsumer::addConsumer(Raffle: [0xe9D52cB94987eB686351B642479C688797A54cb9], VRFCoordinatorV2: [0x8103B0A8A00be2DDC778e6e7eaa21791Cd364625], 73878881164087465189628802004031071946213112452267996662201529735988893001902 [7.387e76], 0x66C6e902bA95dAD75e84152361A210b3c7Dc1e9f)
    │   ├─ [0] console::log("Adding consumer contract: ", Raffle: [0xe9D52cB94987eB686351B642479C688797A54cb9]) [staticcall]
    │   │   └─ ← [Stop] 
    │   ├─ [0] console::log("To vrfCoordinator: ", Raffle: [0xe9D52cB94987eB686351B642479C688797A54cb9]) [staticcall]
    │   │   └─ ← [Stop] 
    │   ├─ [0] console::log("On chainId: ", 11155111 [1.115e7]) [staticcall]
    │   │   └─ ← [Stop] 
    │   ├─ [0] VM::startBroadcast(0x66C6e902bA95dAD75e84152361A210b3c7Dc1e9f)
    │   │   └─ ← [Return] 
    │   ├─ [236] VRFCoordinatorV2::addConsumer(73878881164087465189628802004031071946213112452267996662201529735988893001902 [7.387e76], Raffle: [0xe9D52cB94987eB686351B642479C688797A54cb9])
    │   │   └─ ← [Revert] EvmError: Revert
    │   └─ ← [Revert] EvmError: Revert
    └─ ← [Revert] EvmError: Revert`

== Logs ==
  Adding consumer contract:  0xe9D52cB94987eB686351B642479C688797A54cb9
  To vrfCoordinator:  0x8103B0A8A00be2DDC778e6e7eaa21791Cd364625
  On chainId:  11155111
Error: 
script failed: <empty revert data>
make: *** [Makefile:43: deploy-sepolia] Error 1

My addConsumer:
`contract AddConsumer is Script {
    function addConsumerUsingConfig(address mostRecentDeployment) public {
        HelperConfig helperConfig = new HelperConfig();
        uint256 subId = helperConfig.getConfig().subscriptionId;
        address vrfCoordinator = helperConfig.getConfig().vrfCoordinator;
        address account = helperConfig.getConfig().account;
        addConsumer(mostRecentDeployment, vrfCoordinator, subId, account);
    }

    function addConsumer(address contractToAddtoVrf, address vrfCoordinator, uint256 subId, address account) public {
        console.log("Adding consumer contract: ", contractToAddtoVrf);
        console.log("To vrfCoordinator: ", contractToAddtoVrf);
        console.log("On chainId: ", block.chainid);
        vm.startBroadcast(account);
        VRFCoordinatorV2_5Mock(vrfCoordinator).addConsumer(subId, contractToAddtoVrf);
        vm.stopBroadcast();
    }

    function run() external {
        address mostRecentDeployment = DevOpsTools.get_most_recent_deployment("Raffle", block.chainid);
        addConsumerUsingConfig(mostRecentDeployment);
    }
}`

My HelperConfig:
`contract HelperConfig is CodeConstants, Script {
    error HelperConfig__InvalidChainId();

    struct NetworkConfig {
        uint256 entranceFee;
        uint256 interval;
        address vrfCoordinator;
        bytes32 gasLane;
        uint32 callbackGasLimit;
        uint256 subscriptionId;
        address link;
        address account;
    }

    NetworkConfig public localNetworkConfig;
    mapping(uint256 chainId => NetworkConfig) public networkConfigs;

    constructor() {
        networkConfigs[ETH_SEPOLIA_CHAIN_ID] = getSepoliaEthConfig();
    }

    function getConfigByChainId(uint256 chainId) public returns (NetworkConfig memory) {
        if (networkConfigs[chainId].vrfCoordinator != address(0)) {
            return networkConfigs[chainId];
        } else if (chainId == LOCAL_CHAIN_ID) {
            return getOrCreateAnvilEthConfig();
        } else {
            revert HelperConfig__InvalidChainId();
        }
    }

    function getConfig() public returns (NetworkConfig memory) {
        return getConfigByChainId(block.chainid);
    }

    function getSepoliaEthConfig() public pure returns (NetworkConfig memory) {
        return NetworkConfig({
            entranceFee: 0.01 ether,
            interval: 30, //30 seconds
            vrfCoordinator: 0x8103B0A8A00be2DDC778e6e7eaa21791Cd364625,
            gasLane: 0x8077df514608a09f83e4e8d300645594e5d7234665448ba83f51a50f842bd3d9,
            callbackGasLimit: 500000,
            subscriptionId: 73878881164087465189628802004031071946213112452267996662201529735988893001902,
            link: 0x779877A7B0D9E8603169DdbD7836e478b4624789,
            account: 0x66C6e902bA95dAD75e84152361A210b3c7Dc1e9f
        });
    }

    function getOrCreateAnvilEthConfig() public returns (NetworkConfig memory) {
        //check to see if we set an active network config
        if (localNetworkConfig.vrfCoordinator != address(0)) {
            return localNetworkConfig;
        }
        //Deploy mocks
        vm.startBroadcast();
        VRFCoordinatorV2_5Mock vrfCoordinatorV2_5Mock =
            new VRFCoordinatorV2_5Mock(MOCK_BASE_FEE, MOCK_GAS_PRICE_LINK, MOCK_WEI_PER_UINIT_LINK);
        LinkToken linkToken = new LinkToken();
        vm.stopBroadcast();

        localNetworkConfig = NetworkConfig({
            entranceFee: 0.01 ether,
            interval: 30, //30 seconds
            vrfCoordinator: address(vrfCoordinatorV2_5Mock),
            //doesnt matter
            gasLane: 0x8077df514608a09f83e4e8d300645594e5d7234665448ba83f51a50f842bd3d9,
            callbackGasLimit: 500000,
            subscriptionId: 0,
            link: address(linkToken),
            account: 0x1804c8AB1F12E6bbf3894d4083f33e07309d1f38
        });
        return localNetworkConfig;
    }
}`

SEPOLIA_RPC_URL=https://eth-sepolia.g.alchemy.com/v2/RWESKb0sZ3qf0nMaU6ChhjBY5xWNd9GE

Any Ideas? VRF address seems to be fine

**Comment by szopson** [comment, marked as answer]
Issue was with the wrong vrfCoordinator:

            vrfCoordinator: 0x9DdfaCa8183c41ad55329BdeeD9F6A8d53168B1B,

**Reply by EngrPips** [reply]
Awesome, it's Interesting that you figured that out. Keep crushing it!
